m4/petsc.m4

```diff
@@ -397,7 +397,7 @@ AC_DEFUN([CONFIGURE_PETSC],
           AC_SUBST(MPI_IMPL)
 
           AS_IF([test "x$PETSC_MPI" != x],
-                [AC_DEFINE(HAVE_MPI, 1, [Flag indicating whether or not MPI is available])])
+                [VERSION_MPI])
```

Comment:
Just so I'm clear: we don't run the risk of detecting/using a *different* MPI from the one PETSc is using by running the `VERSION_MPI` check right here, right? Many systems have multiple MPIs installed on them, and we have to be careful not to link against more than one by accident...

Reply:
Since we don't set any variables (except for `HAVE_MPI`) in the `VERSION_MPI`-function, I don't think this leads to problems of actually **linking** against different MPI-libraries.
But the detection of another library than that used by Petsc is of course an issue; in my understanding the `AC_LANG_SAVE` and `AC_LANG_CPLUSPLUS` should take care of this:
1. The the current status is saved
2. The cpp-preferences are loaded (such as `LD-LIBRARY_PATH` etc).
This should avoid cases that wrong libraries are loaded. 

I tried testing this but I always got errors due to compiler-inconsistencies and so the Petsc-dependency was removed.

Reply:
correct me if I am wrong, but at this point it is ensured that we use the same compiler as that used to compile PETSc; this should ensure that we have the same MPI, since it is linked with the compiler.

Moreover, if there is a way to link to the wrong MPI-library in `VERSION_MPI`, than we are using an inconsistent setup anyway; this should be ensured before already.